commands/fetch_templates.go

```
@@ -20,21 +19,22 @@ const DefaultTemplateRepository = "https://github.com/openfaas/templates.git"
 const templateDirectory = "./template/"
 
 // fetchTemplates fetch code templates using git clone.
-func fetchTemplates(templateURL string, refName string, overwrite bool) error {
+func fetchTemplates(templateURL, refName, templateName string, overwrite bool) error {
 	if len(templateURL) == 0 {
 		return fmt.Errorf("pass valid templateURL")
 	}
 
 	dir, err := os.MkdirTemp("", "openfaas-templates-*")
 	if err != nil {
-		log.Fatal(err)
+		return err
```

Better to return an error than use Fatal